pkg/github/github.go

```diff
@@ -7,6 +7,12 @@ import (
 	"golang.org/x/oauth2"
 )
 
+// RepositoriesService provides interface for testing
+type RepositoriesService interface {
+	CreateStatus(ctx context.Context, owner, repo, ref string, status *github.RepoStatus) (*github.RepoStatus, *github.Response, error)
+	GetBranchProtection(ctx context.Context, owner, repo, branch string) (*github.Protection, *github.Response, error)
+}
+
```

Comment:
Interfaces should be defined at the client side. You needed to add `CreateStatus()` to the branch protection mock and `GetBranchProtection()` for the get status mock for no other reason than that you defined the interface at the wrong location. See also https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
The interface is needed in multiple steps ... thus I put it into our custom wrapper for github which looks like a good place for it.
Defining it in one of the steps does not make too much sense to me since then the code will be rather difficult to read.

Where do you suggest to put the interface?

Reply:
> The interface is needed in multiple steps ... thus I put it into our custom wrapper for github which looks like a good place for it.

But that does not seem to be the case. In one step, you expect an interface implementing `CreateStatus()`, in the other step, you expect an interface `GetBranchProtection ()`.  For this reason, to me it makes more sense to define the *expected* interface where you "expect" it, i.e. in the step itself.

> Defining it in one of the steps does not make too much sense to me since then the code will be rather difficult to read.

I don't understand what makes the code more difficult to read. To the contrary, I find it easier to read when I see the interface definition where it is also used.

> Where do you suggest to put the interface?

An interface with only the used function in each step.

Reply:
OK, I see what you mean ...

Two steps now use the `RepositoryService` though ...

Reply:
Ok, but why is that an argument against defining an interface at each step?

Reply:
The big advantage is that if you have a third step which uses a new third method, then you would have to touch the code of two "unrelated" steps. They would indeed be unrelated, if you add the method to the struct only, and define a third interface in the new step.

Reply:
(Of course we have the inconvenience that all steps share the same package, so interface definitions are shared by all steps, even though that isn't really what we mean when defining an interface at a step. So there is the inconvenience of choosing a unique name for all two/three interfaces.)

Reply:
changed it ...